app/src/main/java/systems/v/coldwallet/Activity/ColdWalletActivity.java

```diff
@@ -253,7 +253,13 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                     }
 
                     if (txType != 2 && txType != 3 && txType != 4) {
-                        Toast.makeText(activity, "Incorrect transaction format", Toast.LENGTH_LONG).show();
+                        byte api = Double.valueOf((double)jsonMap.get("api")).byteValue();
+                        if (api > Wallet.API_VERSION) {
```

Comment:
API version check should put out of txType check. The priority of API error is higher than txType error.